src/components/Microsoft.DotNet.UpgradeAssistant/Checks/CanLoadProjectFile.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    public class CanLoadProjectFile : IUpgradeReadyCheck
+    {
+        private readonly ILogger<CanLoadProjectFile> _logger;
+
+        public CanLoadProjectFile(ILogger<CanLoadProjectFile> logger)
+        {
+            _logger = logger;
+        }
+
+        public string Id => nameof(CanLoadProjectFile);
+
+        public Task<bool> IsReadyAsync(IProject project, CancellationToken token)
+        {
+            if (project is null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            try
+            {
+                // Just need to access some values in the file to see if it can be properly loaded.
+                _ = project.GetFile().GetPropertyValue("SomeValue");
+                return Task.FromResult(true);
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception e)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                _logger.LogError("Project {Name} can not be loaded: {Message}", project.FilePath, e.Message);
```

Comment:
We can use `_logger.LogError(e, "Project {Name} can not be loaded", project.FilePath)` to have the logger automatically include the exception information in the log.

Reply:
But then the stack trace gets in the way for the end user. I'd rather keep it at just the message for now.